### ISSUE TITLE
Infer custom spec types from the file location

### ISSUE THREAD

**Issue by perfectritone** [rspec/rspec-rails#1582]
Custom types are already supported by RSpec, but they currently need to be defined in the metadata for each describe block. Once a type has enough examples, this can be tedious and repetitive. 

It would be nice to avoid this by inferring the spec type from the file location like the standard types do.

The current setting still applies
```
config.infer_spec_type_from_file_location!
```
and this change would add
```
config.infer_spec_type_from_file_location!(api_request: %w(spec api_requests))  
```
as well. Now all requests in the spec/api_requests folder will be of type `:api_request`.

**Comment by JonRowe** [comment]
`config.infer_spec_type_from_file_location!` is for legacy spec suites from RSpec `2.x`, it is considered soft deprecated at this point and likely to be removed in RSpec 4, to achieve what you'd like to do try this:

`define_derived_metadata(:file_path => '%r{spec/api_requests}) { |metadata| metadata[:type] ||= :api_request }`

**Comment by michaelckcheung** [comment]
Hi @JonRowe - this post is dated 2 years back. Since then, it appears RSpec has released 3.5, 3.6 and 3.7 with support for automatically inferring the metadata based on the config setting above and the location. Do you know when RSpec will remove support for this, if at all?

In my experience, I've found that the Rails community is often split on this behavior. Some prefer it due to its conformance to the basic principle of convention over configuration, while others appear confused by it.

Any updates/insight you can share on this would be of great help!

**Comment by JonRowe** [comment]
@michaelckcheung it will be removed in RSpec 4. All our generators create the new explicit tagging format.
